### PR TITLE
Fix the syntax error for Nowdoc #4345

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/4345
- Fix the lexers
- Check the closing identifier also when there are two or more newlines before it
- Add unit tests

nb-php-gh-4345-nowdoc-syntax

### CODE REVIEW

**Review by tmysik — APPROVED** [review]
Looks OK to me. @junichi11 let me know if you want to apply my suggestion :) Thanks.

File: php/php.editor/tools/ASTPHP5Scanner.flex [hunk 1]
```diff
@@ -1289,11 +1289,17 @@ NOWDOC_CHARS=({NEWLINE}*(([^a-zA-Z_\x7f-\xff\n\r][^\n\r]*)|({LABEL}[^a-zA-Z0-9_\
...
-    {NEWLINE}{TABS_AND_SPACES}{LABEL}";"?[^\n\r]*[\r\n]? {
-        /* <ST_NOWDOC>{NEWLINE}{TABS_AND_SPACES}{LABEL}";"?[^\n\r]*[\r\n]? */
+    {NEWLINE}+{TABS_AND_SPACES}{LABEL}";"?[^\n\r]*[\r\n]? {
+        /* <ST_NOWDOC>{NEWLINE}+{TABS_AND_SPACES}{LABEL}";"?[^\n\r]*[\r\n]? */
         if (isEndHereOrNowdoc(nowdoc)) {
             String yytext = yytext();
-            int newlineLength = yytext.startsWith("\r\n") ? 2 : 1;
+            int newlineLength = 0;
+            for (int i = 0; i < yylength(); i++) {
+                if (yytext.charAt(i) != '\n' && yytext.charAt(i) != '\r') {
```
[the remark this answers is not in the document]
Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Scanner.flex:1298]: Will change to the following. Thanks!
```java
                char c = yytext.charAt(i);
                if (c != '\n' && c != '\r') {
                    break;
                }
```

**Comment by junichi11** [comment]
@tmysik Will fix it :) Thanks!

File: php/php.editor/tools/ASTPHP5Scanner.flex (hunk 1 above)
[the remark this answers is not in the document]
Comment (junichi11) [reply php/php.editor/tools/ASTPHP5Scanner.flex]: Fixed :)

[the remark this answers is not in the document]
Comment (tmysik) [reply php/php.editor/tools/ASTPHP5Scanner.flex]: Thanks!

**Comment by tmysik** [comment]
@junichi11 Thanks, will merge it once CI finishes.

**Pull request merged**